controller/tests/array_action/ds8k/test_array_mediator_ds8k.py

```diff
@@ -192,17 +194,90 @@ def test_delete_volume_with_flashcopies_as_source_and_target_failed(self):
             Munch({"sourcevolume": "0001",
                    "targetvolume": "0002",
                    "id": "0001:0002",
-                   "state": "valid"
+                   "state": "valid",
+                   "backgroundcopy": "disabled",
+                   "representation": {}
                    }),
             Munch({"sourcevolume": "0003",
                    "targetvolume": "0001",
                    "id": "0003:0001",
-                   "state": "valid"
+                   "state": "valid",
+                   "backgroundcopy": "enabled",
+                   "representation": {}
                    })]
+        self.client_mock.get_flashcopies.return_value = Munch({"out_of_sync_tracks": "0"})
         self.client_mock.delete_volume.side_effect = ClientException("500")
-        with self.assertRaises(array_errors.VolumeDeletionError):
+        with self.assertRaises(array_errors.ObjectIsStillInUseError):
             self.array.delete_volume("0001")
-        self.client_mock.delete_flashcopy.assert_called_once_with("0003:0001")
+
+    def test_delete_volume_with_source_no_flashcopies_deleted(self):
```

Comment:
this name is misleading, the test doesn't assert that no flashcopies were deleted.
did you mean `_raise_object_still_in_use`?

what is the scenario here? is the volume actually a snapshot?

Reply:
done (I changed the name)